intuita-inc/cal.com-demo
# chore: migrate the event-types page to the app directory

## What does this PR do?

This PR migrates the `/event-types` page to the `app` directory (which runs under the App Router).
The migrated page exists under `/future/event-types` and it is accessible at all times.
You might be redirected from `/event-types` to `/future/event-types` under the following circumstances:
1. You pass the `x-calcom-future-routes-override` cookie with a value that converts to true (http-only) to the Next.js server
2. The server set the `x-calcom-future-routes-enabled` cookie beforehand and `APP_ROUTER_EVENT_TYPES_ENABLED` is set to a value that converts to true (this flow is enabled by a percentage of users controlled by the `AB_TEST_BUCKET_PROBABILITY` environment variable).

## Requirement/Documentation
Lighthouse Navigation Desktop Performance: 79 (the pages router) vs 86 (the app router).

## Type of change

## How should this be tested?
There should be no effect on all the other features of the entire application, including navigation.

Path 1:
1. Setup the environment variable `AB_TEST_BUCKET_PROBABILITY` to `100`
2. Setup the environment variable `APP_ROUTER_EVENT_TYPES_ENABLED` to `1`
3 Run the application
4. Navigate to `/event-types`
5. The URL should stay the same but the HTML tag should have the `data-nextjs-router="app"` attribute.
6. The response headers from the `/event-types` HTML page should contain the `Set-Cookie` header with the following data: ```x-calcom-future-routes-enabled=future; Path=/; Expires=Thu, 16 Nov 2023 17:19:29 GMT; HttpOnly```, the expiry time should be now + 30 min.
7. The cookie should be visible in the Application -> Cookies section of the Developer Tools in Chrome.
8. If you refresh the page, the condition under 5 and 7 will be true, but not 6.

Path 2:
1. Setup the environment variable `AB_TEST_BUCKET_PROBABILITY` to `0`
2. Setup the environment variable `APP_ROUTER_EVENT_TYPES_ENABLED` to `1`
3 Run the application
4. Navigate to `/event-types`
5. The URL should stay the same but the HTML tag should NOT have the `data-nextjs-router` attribute.
6. The response headers from the `/event-types` HTML page should contain the `Set-Cookie` header with the following data: ```x-calcom-future-routes-enabled=legacy; Path=/; Expires=Thu, 16 Nov 2023 17:19:29 GMT; HttpOnly```, the expiry time should be now + 30 min.
7. The cookie should be visible in the Application -> Cookies section of the Developer Tools in Chrome.
8. If you refresh the page, the condition under 5 and 7 will be true, but not 6.

## Mandatory Tasks

## Checklist
- I haven't checked if my PR needs changes to the documentation
- I haven't checked if my changes generate no new warnings
